terraso_backend/apps/graphql/schema/groups.py

```diff
@@ -95,3 +95,15 @@ class GroupDeleteMutation(BaseDeleteMutation):
 
     class Input:
         id = graphene.ID()
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        user = info.context.user
+
+        ff_check_permission_on = settings.FEATURE_FLAGS["CHECK_PERMISSIONS"]
+        user_has_delete_permission = user.has_perm(Group.get_perm("delete"), obj=kwargs["id"])
+
+        if ff_check_permission_on and not user_has_delete_permission:
+            raise GraphQLValidationException("User has no permission to delete this data.")
```

Comment:
Will this exception ever be shown to the user? i.e. do we need to localize it?

Reply:
Yes. It might be shown to the user, and we will have to localize. I've been thinking in ways to keep localization only on web client. But I don't think it's the right way to do that. Perhaps we should have localization on API as well.

Reply:
Can we return something like `error.permission.delete` or so?